File: Homework 5/C++/Driver.cpp

#include <iostream>
#include "SingleName.h"
#include "CheckAlphaString.h"
#include "Address.h"
#include "FullName.h"
#include "NameAddress.h"
#include <ctype.h>

using namespace std;

int main() {

	Address address;
	char input[100];
	string street;
	SingleName city;
	SingleName state;
	int zip;
	AlphaString alpha;
	SingleName single;
	CheckAlphaString check;
	SingleName first;
	SingleName middle;
	SingleName last;
	FullName full;
	NameAddress nameaddress;

	cout << "Please enter your first name: ";

	do {

		cin >> input;
		alpha.set(input);
		if (check.check(alpha.get()) == false) {

			cout << "Please do not enter non alphabetic characters." << endl;
			cout << "Please try again: ";

		}

	} while (check.check(alpha.get()) == false);

	first.set(alpha);
	alpha.set("");

	cout << "Please enter your middle name: ";

	do {

		cin >> input;
		alpha.set(input);
		if (check.check(alpha.get()) == false) {

			cout << "Please do not enter non alphabetic characters." << endl;
			cout << "Please try again: ";

		}

	} while (check.check(alpha.get()) == false);

	middle.set(alpha);
	alpha.set("");

	cout << "Please enter your last name: ";

	do {

		cin >> input;
		alpha.set(input);
		if (check.check(alpha.get()) == false) {

			cout << "Please do not enter non alphabetic characters." << endl;
			cout << "Please try again: ";

		}

	} while (check.check(alpha.get()) == false);

	last.set(alpha);

	// All name data collected, set it to the FullName
	// Since setAll() calls induvidual set methods it's a nice all-in-one test
	nameaddress.setName(first, middle, last);


	cout << "Enter a street address (number and name): ";
	cin.ignore();
	cin.getline(input, 100);
	nameaddress.setStreet(input);


	cout << "Enter a city: ";

	do {

		cin >> input;
		alpha.set(input);
		if (check.check(alpha.get()) == false) {

			cout << "Please do not enter non alphabetic characters." << endl;
			cout << "Please try again: ";

		}

	} while (check.check(alpha.get()) == false);

	single.set(alpha);
	nameaddress.setCity(single);
	alpha.set("");

	cout << "Enter a state: ";

	do {

		cin >> input;
		alpha.set(input);
		if (check.check(alpha.get()) == false) {

			cout << "Please do not enter non alphabetic characters." << endl;
			cout << "Please try again: ";

		}

	} while (check.check(alpha.get()) == false);

	single.set(alpha);
	nameaddress.setState(single);
	alpha.set("");

	cout << "Enter a zip-code: ";
	
	do {

		cin >> input;
		if (!address.checkZip(input)) {

			cout << "Please enter numerals only." << endl;
			cout << "Please try again: ";

		}

	} while (!address.checkZip(input));

	zip = stoi(input);

	nameaddress.setZip(zip);

	cout << endl;
	cout << "The full address and name is: \n";
	cout << endl;
	cout << nameaddress.getFullNameSpaces();
	cout << endl;
	cout << nameaddress.getStreet();
	cout << endl;
	cout << nameaddress.getCity().get().get() << ", " << nameaddress.getState().get().get() << " " << nameaddress.getZip();
	cout << endl;
	cout << endl;
	cout << "Your name in \"last, first\" is: " << nameaddress.getLastCommaFirst() << endl;
	cout << endl;
	system("pause");

}

File: Homework 5/Java/Driver.java
/*
 * Test program for group 9, assignment 4
* Yvonne Newman, Paul Durandt, Mario Muscarella and Andrew Yu
* 
* 
* Objective: Test the NameAddressID class by printing the full name, accompanying
* 						address (city, state, zip) and ID number.
*/
import java.util.Scanner; 

public class Driver {

	private static Scanner input;

	public static void main(String[] args) {

		Boolean test = false;
		String s = "";
		String hold;
		String street, city, state;
		int zip, id = 0;
		input = new Scanner(System.in);
		
		while (test == false) {
			System.out.println("Enter a full name without title (first middle last): ");
			s = input.nextLine();
			String[] sa = s.split(" ");
			hold = s;
			hold = hold.replaceAll("\\s+", "");
			// check for alphabetic and right number of words
			if (sa.length == 3 && CheckAlphaString.check(hold) == true && s != "") {
				test = true;
			}
		}

		String[] sa = s.split(" ");
		AlphaString word1 = new AlphaString(sa[0]);
		SingleName first = new SingleName(word1);
		AlphaString word2 = new AlphaString(sa[1]);
		SingleName middle = new SingleName(word2);
		AlphaString word3 = new AlphaString(sa[2]);
		SingleName last = new SingleName(word3);

			System.out.println("Enter a street address (number and name): ");
			street = input.nextLine();
			test = false;

		while (test == false) {
			System.out.println("Enter a city: ");
			s = input.nextLine();
			// check for alphabetic
			if (CheckAlphaString.check(s) == true && s != "") {
				test = true;
			}
		}
			city = s;
			test = false;
			
		while (test == false) {
			System.out.println("Enter a state: ");
			s = input.nextLine();
			// check for alphabetic
			if (CheckAlphaString.check(s) == true && s != "") {
				test = true;
			}
		}
		state = s;
		
			System.out.println("Enter a zipcode: ");
			zip = input.nextInt();
			
			System.out.println("Enter an ID number: ");
			id = input.nextInt();
		
		AlphaString cityAlpha = new AlphaString(city);
		SingleName cityName = new SingleName(cityAlpha);
		AlphaString stateAlpha = new AlphaString(state);
		SingleName stateName = new SingleName(stateAlpha);
		NameAddressID nameAddID = new NameAddressID(street, cityName, stateName, zip, 
				first, middle, last, id);
		
		System.out.printf("%nThe name, full address and ID are: %n" );
		System.out.println( nameAddID.getFullNameSpaces());
		System.out.printf( nameAddID.getStreet() + "%n" );
		System.out.printf( nameAddID.getCity() +  ", " +  nameAddID.getState() + " " + 
				nameAddID.getZip() + "%n");
		System.out.printf("ID: " + nameAddID.getID()  + "%n%n" );

	}
}